miscmodule/showroll_command.go

```diff
@@ -0,0 +1,126 @@
+package miscmodule
+
+import (
+	"math"
+	"math/rand"
+	"regexp"
+	"strconv"
+	"strings"
+
+	bot "../sweetiebot"
+	"github.com/blackhole12/discordgo"
+)
+
+var diceregex = regexp.MustCompile("[0-9]*d[0-9]+")
+
+type showrollCommand struct {
+}
+
+func (c *showrollCommand) Info() *bot.CommandInfo {
+	return &bot.CommandInfo{
+		Name:              "Showroll",
+		Usage:             "Evaluates a dice expression.",
+		ServerIndependent: true,
+	}
+}
+func (c *showrollCommand) eval(args []string, index *int, info *bot.GuildInfo) string {
+	var s string
+	for *index < len(args) {
+		s += value(args, index) + "\n"
+	}
+	return s
+}
+func (c *showrollCommand) value(args []string, index *int, info *bot.GuildInfo) string {
+	*index++
+	if diceregex.MatchString(args[*index-1]) {
+		dice := strings.SplitN(args[*index-1], "d", 2)
+		var multiplier, num, threshold, fail int64 = 1, 1, 0, 0
+		var s string
+		s = "Rolling " + args[*index-1] + ": "
+		if len(dice) > 1 {
+			if len(dice[0]) > 0 {
+				multiplier, _ = strconv.ParseInt(dice[0], 10, 64)
+			}
+			if strings.Contains(dice[1], "t") {
+				tdice := strings.SplitN(dice[1], "t", 2)
+				dice[1] = tdice[0]
+				if strings.Contains(tdice[1], "f") {
+					fdice := strings.SplitN(tdice[1], "f", 2)
+					threshold, _ = strconv.ParseInt(fdice[0], 10, 64)
+					fail, _ = strconv.ParseInt(fdice[1], 10, 64)
+				} else {
+					threshold, _ = strconv.ParseInt(tdice[1], 10, 64)
+				}
+				if threshold == 0 {
+					return s + "Can't roll that! Check dice expression."
+				}
+			}
+			if strings.Contains(dice[1], "f") {
+				fdice := strings.SplitN(dice[1], "f", 2)
+				dice[1] = fdice[0]
+				fail, _ = strconv.ParseInt(fdice[1], 10, 64)
+				if fail == 0 {
+					return s + "Can't roll that! Check dice expression."
+				}
+			}
+			num, _ = strconv.ParseInt(dice[1], 10, 64)
+		} else {
+			num, _ = strconv.ParseInt(dice[0], 10, 64)
+		}
+		if multiplier < 1 || num < 1 {
+			return s + "Can't roll that! Check dice expression."
+		}
+		if multiplier > 9999 {
+			return s + "I don't have that many dice..."
+		}
```

Comment:
This upper limit is alarmingly high. The normal dice roller can have 9999 dice because it just adds them all up, but this will actually try to display _9999 dice!_ At **most** this limit should be 999, but I think a much more reasonable limit would be 250. This limit should also be a constant declared at the top of the file.